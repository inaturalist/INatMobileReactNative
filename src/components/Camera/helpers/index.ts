import { Alert } from "react-native";
import { log } from "../../../../react-native-logs.config";

const logger = log.extend("CameraView");

type Error = {
  message: string;
}

type Event = {
  log: string;
}

const handleClassifierError = (error: Error) => {
  // When we hit this error, there is an error with the classifier.
  logger.info(`handleClassifierError: ${error.message}`);
  Alert.alert("error", error.message);
};

const handleDeviceNotSupported = (error: Error) => {
  // When we hit this error, something with the current device is not supported.
  logger.info(`handleDeviceNotSupported: ${error.message}`);
  Alert.alert("error", error.message);
};

const handleCaptureError = (error: Error) => {
  // When we hit this error, taking a photo did not work correctly
  logger.info(`handleCaptureError: ${error.message}`);
  Alert.alert("error", error.message);
};

const handleCameraError = (error: Error) => {
  // This error is thrown when it does not fit in any of the above categories.
  logger.info(`handleCameraError: ${error.message}`);
  Alert.alert("error", error.message);
};

const handleLog = (event: Event) => {
  logger.info(event.log);
};

export {
  handleClassifierError,
  handleDeviceNotSupported,
  handleCaptureError,
  handleCameraError,
  handleLog,
};
