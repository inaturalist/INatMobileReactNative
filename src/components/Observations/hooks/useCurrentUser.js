// @flow

import inatjs from "inaturalistjs";
import { useEffect, useState } from "react";

import { getJWTToken } from "../../LoginSignUp/AuthenticationService";

const useCurrentUser = ( ): Object => {
  const [currentUser, setCurrentUser] = useState( null );

  useEffect( ( ) => {
    let isCurrent = true;
    const fetchUserProfile = async ( ) => {
      try {
        const apiToken = await getJWTToken( false );
        if ( !apiToken ) {
          isCurrent = false;
          return;
        }
        const options = {
          api_token: apiToken
        };
        const response = await inatjs.users.me( options );
        const { results } = response;
        if ( !isCurrent ) { return; }
        setCurrentUser( results[0].id );
      } catch ( e ) {
        if ( !isCurrent ) { return; }
        console.log( "Couldn't fetch current user:", e.message );
      }
    };

    fetchUserProfile( );
    return ( ) => {
      isCurrent = false;
    };
  }, [] );

  return currentUser;
};

export default useCurrentUser;
