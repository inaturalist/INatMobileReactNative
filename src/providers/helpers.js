// @flow

const USER_FIELDS = {
  icon_url: true,
  id: true,
  login: true,
  name: true
};

const TAXON_FIELDS = {
  default_photo: {
    url: true
  },
  iconic_taxon_name: true,
  name: true,
  preferred_common_name: true,
  rank: true,
  rank_level: true
};

const ID_FIELDS = {
  body: true,
  category: true,
  created_at: true,
  current: true,
  disagreement: true,
  taxon: TAXON_FIELDS,
  updated_at: true,
  // $FlowFixMe
  user: Object.assign( { }, USER_FIELDS, { id: true } ),
  uuid: true,
  vision: true
};

const PHOTO_FIELDS = {
  id: true,
  attribution: true,
  license_code: true,
  url: true
};

const COMMENT_FIELDS = {
  body: true,
  created_at: true,
  id: true,
  user: USER_FIELDS
};

const OBSERVATION_PHOTOS_FIELDS = {
  id: true,
  photo: PHOTO_FIELDS,
  position: true,
  uuid: true
};

const MESSAGE_FIELDS = {
  subject: true,
  body: true,
  from_user: USER_FIELDS,
  to_user: USER_FIELDS
};

const FIELDS = {
  comments: COMMENT_FIELDS,
  created_at: true,
  description: true,
  geojson: true,
  identifications: ID_FIELDS,
  latitude: true,
  location: true,
  longitude: true,
  observation_photos: OBSERVATION_PHOTOS_FIELDS,
  place_guess: true,
  quality_grade: true,
  taxon: TAXON_FIELDS,
  time_observed_at: true,
  user: USER_FIELDS
};

export {
  FIELDS,
  MESSAGE_FIELDS
};
