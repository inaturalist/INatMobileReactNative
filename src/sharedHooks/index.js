export { default as useAuthenticatedMutation } from "./useAuthenticatedMutation";
export { default as useAuthenticatedQuery } from "./useAuthenticatedQuery";
export { default as useCurrentObservationLocation } from "./useCurrentObservationLocation";
export { default as useCurrentUser } from "./useCurrentUser";
export { default as useDebugMode } from "./useDebugMode";
export { default as useDeviceOrientation } from "./useDeviceOrientation";
export { default as useIconicTaxa } from "./useIconicTaxa";
export { default as useInfiniteNotificationsScroll } from "./useInfiniteNotificationsScroll";
export { default as useInfiniteObservationsScroll } from "./useInfiniteObservationsScroll";
export { default as useInfiniteScroll } from "./useInfiniteScroll";
export { default as useInterval } from "./useInterval";
export { default as useIsConnected } from "./useIsConnected";
export { default as useLocalObservation } from "./useLocalObservation";
export { default as useLocalObservations } from "./useLocalObservations";
export { default as useNumUnuploadedObservations } from "./useNumUnuploadedObservations";
export { default as useObservationsUpdates } from "./useObservationsUpdates";
export { default as useObservationUpdatesWhenFocused } from "./useObservationUpdatesWhenFocused";
export { default as useRemoteObservation } from "./useRemoteObservation";
export { default as useShare } from "./useShare";
export { default as useStoredLayout } from "./useStoredLayout";
export { default as useTaxon } from "./useTaxon";
export { default as useTranslation } from "./useTranslation";
export { default as useUserLocation } from "./useUserLocation";
export { default as useUserMe } from "./useUserMe";
