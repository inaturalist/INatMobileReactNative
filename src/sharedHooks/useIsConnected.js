// @flow
import { useNetInfo } from "@react-native-community/netinfo";

const useIsConnected = ( ): boolean => {
  const { isInternetReachable } = useNetInfo( );
  return isInternetReachable;
};

export default useIsConnected;
