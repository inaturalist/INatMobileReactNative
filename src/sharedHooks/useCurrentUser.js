// @flow

import { RealmContext } from "providers/contexts";
import { useMemo } from "react";
import User from "realmModels/User";

const { useRealm } = RealmContext;

const useCurrentUser = ( ): ?Object => {
  const realm = useRealm( );
  const currentUser = User.currentUser( realm );
  const currentUserIsValid = currentUser?.isValid( );
  return useMemo( ( ) => {
    if ( currentUserIsValid ) {
      return currentUser;
    }
    return null;
  }, [
    currentUser,
    currentUserIsValid
  ] );
};

export default useCurrentUser;
