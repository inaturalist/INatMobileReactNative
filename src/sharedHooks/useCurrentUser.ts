import { RealmContext } from "providers/contexts";
import { useMemo } from "react";
import User from "realmModels/User.ts";

const { useRealm } = RealmContext;

const useCurrentUser = ( ): User | null => {
  const realm = useRealm( );
  // TODO there has to be a better way. useMemo isn't doing anything here
  // because currentUser changes every time User.currentUser() gets called,
  // resulting in a lot of unnecessary renders and effects getting
  // triggered... but if you actually memoize the value, you end up
  // with "Accessing object which has been invalidated or deleted" errors
  const currentUser = User.currentUser( realm );
  return useMemo( ( ) => {
    if ( currentUser?.isValid( ) ) {
      return currentUser;
    }
    return null;
  }, [
    currentUser
  ] );
};

export default useCurrentUser;
