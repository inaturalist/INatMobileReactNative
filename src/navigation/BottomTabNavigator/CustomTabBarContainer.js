// @flow
import { useDrawerStatus } from "@react-navigation/drawer";
import type { Node } from "react";
import React, { useMemo, useState } from "react";
import User from "realmModels/User";
import { useCurrentUser, useTranslation } from "sharedHooks";

import CustomTabBar from "./CustomTabBar";

const DRAWER_ID = "OPEN_DRAWER";
const EXPLORE_SCREEN_ID = "RootExplore";
const OBS_LIST_SCREEN_ID = "ObservationsStackNavigator";
const NOTIFICATIONS_SCREEN_ID = "Notifications";

type Props = {
  navigation: Object
};

const CustomTabBarContainer = ( { navigation }: Props ): Node => {
  const { t } = useTranslation( );
  const currentUser = useCurrentUser( );
  const [activeTab, setActiveTab] = useState( OBS_LIST_SCREEN_ID );
  const isDrawerOpen = useDrawerStatus() === "open";

  const tabs = useMemo( ( ) => ( [
    {
      icon: "hamburger-menu",
      testID: DRAWER_ID,
      accessibilityLabel: t( "Open-drawer" ),
      accessibilityHint: t( "Opens-the-side-drawer-menu" ),
      size: 32,
      onPress: ( ) => {
        navigation.openDrawer( );
        setActiveTab( null );
      },
      active: isDrawerOpen
    },
    {
      icon: "compass-rose-outline",
      testID: EXPLORE_SCREEN_ID,
      accessibilityLabel: t( "Navigate-to-explore-screen" ),
      accessibilityHint: t( "Navigates-to-explore" ),
      size: 40,
      onPress: ( ) => {
        navigation.navigate( "RootExplore" );
        setActiveTab( EXPLORE_SCREEN_ID );
      },
      active: EXPLORE_SCREEN_ID === activeTab
    },
    {
      icon: "person",
      userIconUri: User.uri( currentUser ),
      testID: "NavButton.personIcon",
      accessibilityLabel: t( "Observations" ),
      accessibilityHint: t( "Navigates-to-observations" ),
      size: 40,
      onPress: ( ) => {
        navigation.navigate( "ObservationsStackNavigator", {
          screen: "ObsList"
        } );
        setActiveTab( OBS_LIST_SCREEN_ID );
      },
      active: OBS_LIST_SCREEN_ID === activeTab
    },
    {
      icon: "notifications-bell",
      testID: NOTIFICATIONS_SCREEN_ID,
      accessibilityLabel: t( "Notifications" ),
      accessibilityHint: t( "Navigates-to-notifications" ),
      size: 32,
      onPress: ( ) => {
        navigation.reset( {
          index: 0,
          routes: [
            {
              name: "ObservationsStackNavigator",
              params: {
                screen: "Notifications"
              }
            }
          ]
        } );
        setActiveTab( NOTIFICATIONS_SCREEN_ID );
      },
      active: NOTIFICATIONS_SCREEN_ID === activeTab
    }
  ] ), [
    activeTab,
    currentUser,
    isDrawerOpen,
    navigation,
    t
  ] );

  return (
    <CustomTabBar
      tabs={tabs}
    />
  );
};

export default CustomTabBarContainer;
