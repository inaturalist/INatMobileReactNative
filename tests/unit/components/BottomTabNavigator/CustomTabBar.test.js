import { screen } from "@testing-library/react-native";
import CustomTabBarContainer from "navigation/BottomTabNavigator/CustomTabBarContainer";
import React from "react";
import * as useCurrentUser from "sharedHooks/useCurrentUser";
import * as useIsConnected from "sharedHooks/useIsConnected.ts";
import factory from "tests/factory";
import faker from "tests/helpers/faker";
import { renderComponent } from "tests/helpers/render";

const mockUser = factory( "LocalUser", {
  login: faker.internet.userName( ),
  icon_url: faker.image.url( )
} );

jest.mock( "sharedHooks/useCurrentUser", ( ) => ( {
  __esModule: true,
  default: () => undefined
} ) );

jest.mock( "sharedHooks/useAuthenticatedQuery", () => ( {
  __esModule: true,
  default: () => ( {
    data: 0
  } )
} ) );

describe( "CustomTabBar", () => {
  beforeEach( ( ) => {
    jest.useFakeTimers();
  } );

  it( "should render correctly", async () => {
    renderComponent( <CustomTabBarContainer navigation={jest.fn( )} /> );

    await expect( screen ).toMatchSnapshot();
  } );

  it( "should not have accessibility errors", async () => {
    const tabBar = <CustomTabBarContainer navigation={jest.fn( )} />;

    await expect( tabBar ).toBeAccessible();
  } );

  it( "should display person icon while user is logged out", async () => {
    renderComponent( <CustomTabBarContainer navigation={jest.fn( )} isOnline /> );

    const personIcon = screen.getByTestId( "NavButton.personIcon" );
    await expect( personIcon ).toBeVisible( );
  } );

  it( "should display avatar while user is logged in", async () => {
    jest.spyOn( useCurrentUser, "default" ).mockImplementation( () => mockUser );
    renderComponent( <CustomTabBarContainer navigation={jest.fn( )} isOnline /> );

    const avatar = screen.getByTestId( "UserIcon.photo" );
    await expect( avatar ).toBeVisible( );
  } );

  it( "should display person icon when connectivity is low", async ( ) => {
    jest.spyOn( useIsConnected, "default" ).mockImplementation( () => false );
    renderComponent( <CustomTabBarContainer navigation={jest.fn( )} isOnline={false} /> );

    const personIcon = screen.getByTestId( "NavButton.personIcon" );
    await expect( personIcon ).toBeVisible( );
  } );
} );
