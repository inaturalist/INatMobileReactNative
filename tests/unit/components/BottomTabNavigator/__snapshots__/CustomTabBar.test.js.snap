// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomTabBar should render correctly 1`] = `
<View
  collapsable={false}
  pointerEvents="box-none"
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    className="flex-1"
    style={
      {
        "flex": 1,
      }
    }
  >
    <View
      accessibilityRole="tablist"
      style={
        [
          [
            {
              "flexDirection": "row",
            },
            {
              "backgroundColor": "#ffffff",
            },
            {
              "justifyContent": "space-evenly",
            },
            {
              "alignItems": "center",
            },
            {
              "paddingBottom": 4,
              "paddingLeft": 4,
              "paddingRight": 4,
              "paddingTop": 4,
            },
            {
              "marginBottom": 0,
              "marginLeft": 0,
              "marginRight": 0,
              "marginTop": 0,
            },
            {
              "elevation": 20,
              "shadowColor": "#000000",
              "shadowOffset": {
                "height": -3,
                "width": 0,
              },
              "shadowOpacity": 0.2,
              "shadowRadius": 2,
            },
          ],
        ]
      }
    >
      <View
        accessibilityHint="Opens the side drawer menu."
        accessibilityLabel="Open drawer"
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "alignItems": "center",
              "height": 44,
              "justifyContent": "center",
              "opacity": 1,
              "width": 44,
            },
            false,
            undefined,
          ]
        }
        testID="OPEN_DRAWER"
      >
        <View
          style={
            [
              [
                {
                  "position": "relative",
                },
                {
                  "position": "relative",
                },
                {
                  "left": 0.2,
                },
                {
                  "top": 0.1,
                },
              ],
            ]
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": "#454545",
                  "fontSize": 32,
                },
                undefined,
                {
                  "fontFamily": "INatIcon",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
          >
            
          </Text>
        </View>
      </View>
      <View
        accessibilityHint="Navigates to explore."
        accessibilityLabel="Navigate to explore screen"
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "alignItems": "center",
              "height": 44,
              "justifyContent": "center",
              "opacity": 1,
              "width": 44,
            },
            false,
            undefined,
          ]
        }
        testID="RootExplore"
      >
        <View
          style={
            [
              [
                {
                  "position": "relative",
                },
                {
                  "position": "relative",
                },
                {
                  "left": 0.2,
                },
                {
                  "top": 0.1,
                },
              ],
            ]
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": "#454545",
                  "fontSize": 40,
                },
                undefined,
                {
                  "fontFamily": "INatIcon",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
          >
            
          </Text>
        </View>
      </View>
      <Modal
        animationType="none"
        deviceHeight={null}
        deviceWidth={null}
        hardwareAccelerated={false}
        hideModalContentWhileAnimating={true}
        onBackdropPress={[Function]}
        onModalHide={[Function]}
        onModalWillHide={[Function]}
        onModalWillShow={[Function]}
        onRequestClose={[Function]}
        onSwipeComplete={[Function]}
        panResponderThreshold={4}
        scrollHorizontal={false}
        scrollOffset={0}
        scrollOffsetMax={0}
        scrollTo={null}
        statusBarTranslucent={false}
        supportedOrientations={
          [
            "portrait",
            "landscape",
          ]
        }
        swipeDirection="down"
        swipeThreshold={100}
        transparent={true}
        useNativeDriverForBackdrop={true}
        visible={false}
      />
      <View
        accessibilityHint="Opens add observation modal."
        accessibilityLabel="Add observations"
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            [
              {
                "width": 69,
              },
              {
                "height": 69,
              },
              {
                "borderBottomLeftRadius": 9999,
                "borderBottomRightRadius": 9999,
                "borderTopLeftRadius": 9999,
                "borderTopRightRadius": 9999,
              },
              {
                "overflow": "hidden",
              },
              {
                "elevation": 5,
                "shadowColor": "#000000",
                "shadowOffset": {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.25,
                "shadowRadius": 2,
              },
            ],
          ]
        }
        testID="add-obs-button"
      >
        <BVLinearGradient
          angle={156.95}
          colors={
            [
              4285836288,
              4280909703,
            ]
          }
          endPoint={
            {
              "x": 0.5,
              "y": 1,
            }
          }
          locations={null}
          startPoint={
            {
              "x": 0.5,
              "y": 0,
            }
          }
          useAngle={true}
        >
          <View
            style={
              [
                [
                  {
                    "flexGrow": 1,
                  },
                  {
                    "aspectRatio": 1,
                  },
                  {
                    "display": "flex",
                  },
                  {
                    "alignItems": "center",
                  },
                  {
                    "justifyContent": "center",
                  },
                ],
              ]
            }
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                [
                  {
                    "color": "#ffffff",
                    "fontSize": 31,
                  },
                  undefined,
                  {
                    "fontFamily": "INatIcon",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  {},
                ]
              }
            >
              
            </Text>
          </View>
        </BVLinearGradient>
      </View>
      <View
        accessibilityHint="Navigates to observations."
        accessibilityLabel="Observations"
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "alignItems": "center",
              "height": 44,
              "justifyContent": "center",
              "opacity": 1,
              "width": 44,
            },
            false,
            undefined,
          ]
        }
        testID="NavButton.personIcon"
      >
        <View
          style={
            [
              [
                {
                  "position": "relative",
                },
                {
                  "position": "relative",
                },
                {
                  "left": 0.2,
                },
                {
                  "top": 0.1,
                },
              ],
            ]
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": "#74AC00",
                  "fontSize": 40,
                },
                undefined,
                {
                  "fontFamily": "INatIcon",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
          >
            
          </Text>
        </View>
      </View>
      <View
        accessibilityHint="Navigates-to-notifications"
        accessibilityLabel="Notifications"
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "alignItems": "center",
              "height": 44,
              "justifyContent": "center",
              "opacity": 1,
              "width": 44,
            },
            false,
            undefined,
          ]
        }
        testID="Notifications"
      >
        <View
          style={
            [
              [
                {
                  "position": "relative",
                },
                {
                  "position": "relative",
                },
                {
                  "left": 0.2,
                },
                {
                  "top": 0.1,
                },
              ],
            ]
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": "#454545",
                  "fontSize": 32,
                },
                undefined,
                {
                  "fontFamily": "INatIcon",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
